python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -877,9 +877,17 @@ def info(self, full=False):
                 s += ", %i other concurrent sessions:\n * %s" % (len(other_sessions), "\n * ".join(other_sessions))
         return s
 
+    def other_session_active(self):
+        """ Returns a True/False for if other sessions have been detected from the Repository """
+        if self.repository.get_other_sessions():
+            return True
+        else:
+            return False
+
     def print_other_sessions(self):
-        other_sessions = self.repository.get_other_sessions()
-        if len(other_sessions) > 0:
+        """ Prints a warning if other active sessions have been detected """
+        if self.other_session_active():
+            other_sessions = self.repository.get_other_sessions()
```

Comment:
I think we have a strong disagreement about what it means to simplify the API in this case I don't think the registry code need know what external instances there are only that there are some. I don't think this information should be passed up from the SessionLock as at the moment it's very implementation specific. I can see that ```def other_session_active(self):``` can be simplified but I don't think we should be passing knowledge of other session locks outside of the sessionlock code.